esphome/components/mt6701/mt6701_sensor.cpp

```diff
@@ -0,0 +1,48 @@
+#include "mt6701_sensor.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace mt6701 {
+
+static const char *const TAG = "mt6701";
+static const uint8_t REGISTER_ANGLE = 0x03;
+
+float position2degrees(uint16_t pos) {
+  const uint16_t resolution = 0x4000;  // 14 bits
+  return (pos * 360.0f) / resolution;
+}
+
+void MT6701Sensor::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up MT6701 '%s'...", this->name_.c_str());
+
+  if (!read_position().has_value()) {
+    this->mark_failed();
+    return;
+  }
+}
+
+void MT6701Sensor::dump_config() {
+  LOG_SENSOR("", "MT6701", this);
+  LOG_UPDATE_INTERVAL(this);
+  LOG_I2C_DEVICE(this);
+}
+
+void MT6701Sensor::update() {
+  auto pos = read_position();
+  if (!pos.has_value()) {
+    this->status_set_warning();
```

Comment:
```suggestion
    this->status_set_warning(READ_FAILED_MESSAGE);
```

This will print warning message only if component does not have warning status set. It will not spam in log if that is your concern.